### PR TITLE
Bug/cs 43011 dependency issue

### PR DESCRIPTION
**Pull request by abhinav-from-contentstack**
- resolved dependency issues
- added allowJs: true to tsconfig.json for warnings generated while running unit tests
- changed objectHashIgnoreUnknownHack to false
- Added changes from [Deepak-Kharah:VC-218-multiple-array-support](https://github.com/Deepak-Kharah/contentstack-utils-javascript/tree/VC-218-multiple-array-support)

### CODE REVIEW

File: src/entry-editable.ts [hunk 1]
```diff
@@ -8,27 +8,91 @@ export function addTags(entry: EntryModel, contentTypeUid: string, tagsAsObject:
...
     const tags: any = {}
     Object.entries(content).forEach(([key, value]) => {      
+        if (key === '$') return
+
         switch (typeof value) {
             case "object":
                 if (Array.isArray(value)) {
                     value.forEach((obj, index) => {
+                        const childKey = `${key}__${index}`
+                        const parentKey = `${key}__parent`
```
Comment (ishaileshmishra) [inline src/entry-editable.ts:18]: Why this comment is under the code ?

File: src/entry-editable.ts [hunk 2]
```diff
@@ -8,27 +8,91 @@ export function addTags(entry: EntryModel, contentTypeUid: string, tagsAsObject:
...
+                         *      "array": {"data-cslp": "content_type_uid.entry_uid.locale.array"}
+                         *      "array__0": {"data-cslp": "content_type_uid.entry_uid.locale.array.0"}
+                         *      "array__1": {"data-cslp": "content_type_uid.entry_uid.locale.array.1"}
+                         *  }
+                         * }
+                         */
+                        tags[childKey] = getTagsValue(`${prefix}.${key}.${index}`, tagsAsObject)
+                        tags[parentKey] = getParentTagsValue(`${prefix}.${key}`, tagsAsObject)
                         if (typeof obj !== 'undefined' && obj !== null && obj._content_type_uid !== undefined && obj.uid !== undefined) {
+                            /**
```
Comment (ishaileshmishra) [inline src/entry-editable.ts:33]: this also @abhinav-from-contentstack

File: src/entry-editable.ts (hunk 1 above)
Comment (abhinav-from-contentstack) [reply src/entry-editable.ts:18]: These are Deepak's changes @ishaileshmishra , I added them as is.

File: src/entry-editable.ts [hunk 3]
```diff
@@ -8,27 +8,91 @@ export function addTags(entry: EntryModel, contentTypeUid: string, tagsAsObject:
...
+                             */
                             value[index].$ = getTag(obj, `${obj._content_type_uid}.${obj.uid}.${obj.locale || locale}`, tagsAsObject, locale)
-                        }else {
-                            if (typeof obj === "object") {
-                                obj.$ = getTag(obj, `${prefix}.${key}.${index}`, tagsAsObject, locale)
-                            } else {
-                                tags[key] = getTagsValue(`${prefix}.${key}`, tagsAsObject)
-                            }
+                        }else if (typeof obj === "object") {
+                            /**
```
Comment (ishaileshmishra) [inline src/entry-editable.ts:46]: here @abhinav-from-contentstack

**Review by ishaileshmishra — COMMENTED** [review]
are these code comment needed in the master @abhinav-from-contentstack ?

File: src/entry-editable.ts (hunk 1 above)
Comment (abhinav-from-contentstack) [reply src/entry-editable.ts:18]: I'll remove and recommit. @ishaileshmishra

**Comment by abhinav-from-contentstack** [comment]
@ishaileshmishra, I don't see any harm in keeping these comments. They explain the functioning of the code. But, if you'd like we can omit them from `master` branch

**Comment by abhinav-from-contentstack** [comment]
As the code is public anyways, it'll help anyone who would like to contribute.

**Review by nadeem-cs — APPROVED** [review]

File: src/entry-editable.ts (hunk 2 above)
Comment (abhinav-from-contentstack) [reply src/entry-editable.ts:33]: keeping the comments as they help understanding code function

File: src/entry-editable.ts (hunk 3 above)
Comment (abhinav-from-contentstack) [reply src/entry-editable.ts:46]: keeping the comments as they help understanding code function

**Pull request merged**